torchci/components/LogViewer.tsx

```diff
@@ -266,3 +321,40 @@ export default function LogViewer({
     </div>
   );
 }
+
+export function SearchLogViewer({
+  url,
+  lines,
+}: {
+  url: string;
+  lines: [number[], string[]] | undefined;
+}) {
+  // This is the search equivalent of LogVeiwer.  It is almost the same thing
+  // but also displays info about how many matches there were for the search
+
+  let infoString = "";
+  let lineNumbers: number[] = [];
+  let lineTexts: string[] = [];
+  if (lines === undefined) {
+    lineNumbers = [];
+    lineTexts = [];
+    infoString = "Failed to retrieve log";
```

Comment:
Nit:  Is there a way to put up a more benign message for pending jobs as their logs are not available on S3 yet?

![Screenshot 2024-01-01 at 20 26 38](https://github.com/pytorch/test-infra/assets/475357/2767f777-44d4-438a-8ef9-e70cae8807ee)